addons/account/wizard/account_move_send.py

```diff
@@ -597,11 +597,10 @@ def _generate_invoice_fallback_documents(self, invoices_data):
         :param invoices_data:   The collected data for invoices so far.
         """
         for invoice, invoice_data in invoices_data.items():
-            form = invoice_data['_form']
-            if form._need_invoice_document(invoice) and invoice_data.get('error'):
+            if self._need_invoice_document(invoice) and invoice_data.get('error'):
```

Comment:
why reverting this change?
It is needed for multi-record send

Reply:
Strictly speaking I don't think it's necessary, if you look at `_need_invoice_document` and `_hook_invoice_document_after_pdf_report_render` they could both be switched to `@api.model` methods (which we have in 17.0 for all hooks).

But I get your point, will do it for consistency and possible (even if very unlikely) future overrides in 16.4. :thinking: :+1:

Reply:
In multi mode `self.invoice_mode` is `invoice_multi` and we don't get `proforma_pdf_attachment` data filled
https://github.com/odoo/odoo/blob/b78412af4e3157fa81bd03514cf5f14517e7a7f8/addons/account/wizard/account_move_send.py#L297
while `form.invoice_mode` is `invoice_single`
But now that I look at it a bit more, it seems a bit hack-ish, perhaps it was supposed to be this way :thinking:

Reply:
Oh indeed, it was definitely hacky, we removed all that form overhead in next version :)